tasks/retrieval_deal.go

```diff
@@ -219,7 +219,35 @@ func (de *retrievalDealExecutor) executeAndMonitorDeal(ctx context.Context, upda
 				if event.Event == retrievalmarket.ClientEventDealAccepted {
 					stage = RetrievalStageDealAccepted
 					dealStage = RetrievalStages[stage]()
-					err := updateStage(ctx, stage, dealStage)
+					//there should be a deal now.
+					allRetrievals, err := de.node.ClientListRetrievals(ctx)
```

Comment:
are there other things to match on here in pulling out the deal from the retrieval? cc @hannahhoward